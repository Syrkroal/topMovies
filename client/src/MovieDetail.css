.movie-detail-container-center {
    background-color: #282c34;
    width: 100%;
    height: 100%;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    /* align-items: center; */
}

.movie-detail-container {
    background-color: #282c34;
    width: 100%;
    height: 100%;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    /* align-items: center; */
}

.movie-detail-info-container {
    margin: 20px;
    /* width: 100%; */
    /* height: 100%; */
    color: white;
    display: flex;
    justify-content: center;
}

.movie-detail-info {
    width: 50%;
    min-width: 300px;
    margin-left: 20px;
    display: flex;
    flex-direction: column;
}

.movie-detail-header {
    /* font-weight: bold; */
    flex-flow: column wrap;
    color: white;
}

.movie-detail-img {
    height: 500px;
    border-radius: 10px;
}

.movie-detail-title {
    font-size: 40px;
    margin: 20px 0 20px 0;
}

.movie-date-genres {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.movie-detail-actors {
    margin: 0 20px 20px 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.actors-title {
    font-weight: bold;
    font-size: 35px;
}

.actors-container {
    overflow-x: scroll;
    display: flex;
    flex-direction: row;
    width: 80%;
}

.buy-button {
    display: flex;
    justify-content: center;
    align-items: center;

    border-radius: 20px;
    height: 50px;
    width: 200px;
    background-color: #141414;
    cursor: pointer;
    font-weight: bold;
}

.bold {
    font-weight: bold;
}
