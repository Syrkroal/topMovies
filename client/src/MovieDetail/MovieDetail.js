import { useState, useEffect } from "react";
import { useLocation } from "react-router-dom";
import axios from "axios";

import "../App.css";
import "./MovieDetail.css";
import ActorCard from "../ActorCard/ActorCard";

const imgPath = "https://image.tmdb.org/t/p/w500/";

function MovieDetail() {
    const location = useLocation();
    const movie = location.state.movie;

    const [detail, setDetail] = useState(null);
    const [actors, setActors] = useState(null);
    const [genres, setGenres] = useState("");

    useEffect(() => {
        if (!detail) getMovieDetail();
        if (!actors) getMovieActors();
    });

    async function getMovieDetail() {
        const response = await axios.get(
            "http://localhost:4000/movies/info/" + movie.id
        );
        if (response.data.error) return false;

        setDetail(response.data.infos);
        let tmp = "";
        for (let i = 0; i < response.data.infos.genres.length; i++) {
            tmp += response.data.infos.genres[i].name + ", ";
        }
        setGenres(tmp.slice(0, -2));

        return true;
    }

    async function getMovieActors() {
        const response = await axios.get(
            "http://localhost:4000/movies/cast/" + movie.id
        );
        if (response.data.error) return false;
        setActors(response.data.cast);
        console.log(response.data.cast.length);

        return true;
    }

    function buyTicket() {
        return;
    }

    return (
        <div className="movie-detail-container-center">
            <div className="movie-detail-container">
                <div className="movie-detail-info-container">
                    <img
                        className="movie-detail-img"
                        src={imgPath + movie.poster_path}
                        alt="Poster"
                    />
                    <div className="movie-detail-info">
                        <div className="movie-detail-header">
                            <p className="movie-detail-title bold">
                                {movie.title}
                            </p>
                            <div className="movie-date-genres">
                                <p>Released: {movie.release_date}</p>
                                <p>Genres: {genres}</p>
                            </div>
                            <p className="bold">Rating</p>
                            <p>{movie.vote_average} / 10</p>
                            <p className="bold">Overview</p>
                            <p>{movie.overview}</p>
                        </div>
                        <div className="buy-button" onClick={buyTicket}>
                            Purchase Ticket
                        </div>
                    </div>
                </div>

                <div className="movie-detail-actors">
                    <div className="actors-title">Actors</div>
                    {actors && actors.length > 0 && (
                        <div className="actors-container">
                            {actors.map((actor, index) => {
                                return <ActorCard key={index} actor={actor} />;
                            })}
                        </div>
                    )}
                    {actors &&
                        actors.length === 0 &&
                        "No actors for this movie"}
                </div>
            </div>
        </div>
    );
}

export default MovieDetail;
