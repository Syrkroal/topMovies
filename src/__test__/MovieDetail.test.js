import React from "react";
import Enzyme, { shallow } from "enzyme";
import MovieDetail from "../MovieDetail/MovieDetail";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
    useLocation: jest.fn().mockReturnValue({
        state: {
            movie: {
                id: "1",
                title: "title",
                overview: "overview",
                vote_average: "5",
            },
        },
    }),
}));

describe("MovieDetail to match snapshot", () => {
    const container = shallow(<MovieDetail />);
    it("should match the snapshot", () => {
        expect(container.html()).toMatchSnapshot();
    });
});
