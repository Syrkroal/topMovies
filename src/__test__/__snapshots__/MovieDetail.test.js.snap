// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail to match snapshot should match the snapshot 1`] = `"<div class=\\"movie-detail-container-center\\"><div class=\\"movie-detail-container\\"><div class=\\"movie-detail-info-container\\"><img class=\\"movie-detail-img\\" src=\\"https://image.tmdb.org/t/p/w500/undefined\\" alt=\\"Poster\\"/><div class=\\"movie-detail-info\\"><div class=\\"movie-detail-header\\"><p class=\\"movie-detail-title bold\\">title</p><div class=\\"movie-date-genres\\"><p>Released: </p><p>Genres: </p></div><p class=\\"bold\\">Rating</p><p>5 / 10</p><p class=\\"bold\\">Overview</p><p>overview</p></div><div class=\\"buy-button\\">Purchase Ticket</div></div></div><div class=\\"movie-detail-actors\\"><div class=\\"actors-title\\">Actors</div></div></div></div>"`;
